src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/DynamicTable.cs

```diff
@@ -89,11 +88,22 @@ public void Resize(int maxSize)
                 _buffer = newBuffer;
                 _removeIndex = 0;
                 _insertIndex = _count;
-                _maxSize = maxSize;
             }
-            else
+
+            var entry = new HeaderField(staticTableIndex, name, value);
+            _buffer[_insertIndex] = entry;
+            _insertIndex = (_insertIndex + 1) % _buffer.Length;
```

Comment:
Would this be faster and equivalent as:
```C#
_insertIndex++;
if (_insertIndex == _buffer.Length)
{
    _insertIndex = 0;
}
```
or does it not matter?

Reply:
It's slightly faster (~0.5ns).
If we wanted to go after more per-header costs, we should probably look at places like these
https://github.com/dotnet/runtime/blob/8e492e4ff50aa6cf4317faf57a39dd5e4a597504/src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/HeaderField.cs#L22-L28